modules/nextflow/src/main/groovy/nextflow/executor/FluxExecutor.groovy

```diff
@@ -61,7 +61,12 @@ class FluxExecutor extends AbstractGridExecutor {
         List<String> result = ['flux', 'mini', 'submit']
         result << '--setattr=cwd=' + quote(task.workDir)
         result << '--job-name="' + getJobNameFor(task) + '"'
-        result << '--output=' + quote(task.workDir.resolve(TaskRun.CMD_LOG))  // -o OUTFILE
+
+        // Only write output to file if user doesn't want written entirely to terminal
+        Boolean terminal_output = task.config.navigate('terminal_output') as Boolean
```

Comment:
I'm not sure what the best way to handle a "this could be completely undefined, or defined and a boolean" logic. I first tried checking if the value (as a string) was == 'true' and this worked when it was defined in the config, and threw an error in testing here (not defined). I suspect doing a verbatim `task.config.terminal_output == true` would do the same. So I saw this other pattern to "navigate" but I'm not sure that is safe either - I'm wondering if I need something like https://www.logicbig.com/tutorials/misc/groovy/safe-navigation-operator.html. I'm hoping the above is smart enough to convert whatever it finds to boolean!

Reply:
I think we have already `process.debug = true` for this

Reply:
But what I’d they want debug level logging but to the output file? Does debug just control where it writes and it was erroneous of me to not use it to direct the output?

Reply:
And to clarify - when I tested `process.debug` here I saw a _ton_ of output in the main process thread (which for me is in the web server logs) and I definitely don't want that. I want the same output in the logs as currently is there (just the log heading and showing the jobs running) but then the robust output in the process stdout/stderr. So I'm not sure that using process.debug is what I want -it seems to do a bit more.

Reply:
So what I'd like to do is:

1. Ensure that there is some flag that tells the FluxExecutor not to specify any output to a file (this is the functionality I need to see the logs in the UI) e.g.,

![image](https://user-images.githubusercontent.com/814322/203397890-b2638720-36d5-40a7-bf1e-3f494bb5bc6a.png)

2. Also ensure the main process isn't getting overly verbose (it will clutter up the web server where it's running).

I'm happy to test out any suggestions / make any changes that are desired! In practice I couldn't get anything with `process.debug` to work as I wanted, and if I use it here, I suspect we are going to not satisfy point 2 above.

Reply:
Nextflow commonly uses these files for task logs: 
1. `.command.out`: nothing more the task `/dev/stdout`
2. `.command.err`: as above, the task `/dev/stdout` 
3. `.command.log`: the task output as returned by the job scheduler, essentially this file contains the task stdout + stderr + some prolog or epilog depending the batch scheduler. 

When you specify `process.debug = true`. Nextflow reads the content of `.command.out` end print it to the main process stdout (if it's missing fallback to .command.err, if also this is missing fallback to .command.log) 

Don't know exactly how Flux works at this regard, but commonly when the task output log is not specified the stdout/err is written to a file (or a couple of files) whose name is chosen by the file system. This should not alter how the job output is rendered by nextlow. 

I'd suggest to focus this PR on fixing the job status warning.

Reply:
Yep, and I don't (can't) read from a file. I need the output and error in the stdout/stderr of the worker, and _not_ the main process.  So using process.debug isn't going to work. I need a way to control the `--output`, as I've suggested in the PR here.

Reply:
Why won't you let me customize the executor? I don't understand. There is a use case that NextFlow does not work for. As I've stated, I cannot read an output file. I also don't want output in the main process - that's going to a server log that a user won't see. The user will see the output for the process, whatever it is (this is not specific to Nextflow, it's for any thing that is being run). With this tiny addition of a config variable, I'm able to have a beautiful table of process runs with logs specific to them in the UI. Without, the logs are all empty and it's useless.

If you don't like this implementation of an extra option, I can do a hack to "throw it" into the catch all extra arguments param, but then I'd need to loop over it, check for one, and set the variable there. Let me know if this is acceptable and I'm happy to test this out.

Reply:
Hi, PR reviews is all about sharing knowledge and giving advice. I'm not trying to stop customizing the executor, that's good!

Now, I understand you need to control having or not having the flux output log file. 

From Nextflow point is better to have that file, because it can provide useful debugging information to the user if the job fails and for some reason the .commnad.out/err have not been created. So my suggestion is to keep it enabled by default, and add an option to disable it. 

Regarding how to config it. The `process.something` settings are meant to be \*portable* task directives i.e. that applies to all or at least some executors e.g. `cpus`, `memory`, `label`, etc. We tend to add a disk directive only for one executor. 

However, it's perfectly fine to have a `flux { yourSetting = 'here' }` configuration scope in the nextflow config for the option related to Flux (we use camelCase convention for config setting instead of `_` separator). 

I think this should accommodate your needs.

Reply:
Ahh fantastic! So… this is exactly what I had in the first version of the PR with the process.terminal_output flag (false or not present by default leading to printing to output). I thankfully saved that work to a different branch so I’ll see if I can use it to update the PR here, and worst case open a new one.

Thank you! 🙏

Reply:
okay I think that worked! I cloned the repository again (and fetched the different branch with my backup) and then checked out this branch from that branch, and force pushed.

Reply:
oh I missed the camel case though! Let me fix that.

Reply:
okay:

1. rebased
2. changed to camelCase
3. added a test for terminalOutput

I didn't get to run the test locally (my DevContainers session bonked out) but maybe I'll luck out and get it right anyway!